File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque
_input = sys.stdin.readline

n, m, v = map(int, _input().split())
rep = [[] for _ in range(n + 1)]
visited_dfs = [False for _ in range(n + 1)]
visited_bfs = [False for _ in range(n + 1)]
for _ in range(m):
    a, b = map(int, _input().split())
    rep[a].append(b)
    rep[a].sort()
    rep[b].append(a)
    rep[b].sort()
result_dfs = []
result_bfs = []


def dfs(next_item):
    if visited_dfs[next_item]:
        return
    result_dfs.append(str(next_item))
    visited_dfs[next_item] = True
    for e in rep[next_item]:
        if not visited_dfs[e]:
            dfs(e)
    return

def bfs(start):
    visited_bfs[start] = True
    queue = deque([start])
    while queue:
        vt = queue.popleft()
        result_bfs.append(str(vt))
        for e in rep[vt]:
            if not visited_bfs[e]:
                queue.append(e)
                visited_bfs[e] = True
    return


dfs(v)
bfs(v)

print(' '.join(result_dfs))
print(' '.join(result_bfs))

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {

	static boolean[] visitedBfs;
	static boolean[] visitedDfs;
	static List<Integer>[] node;
	static int N;
	static int M;
	static int V;

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());

		N = Integer.parseInt(st.nextToken());
		M = Integer.parseInt(st.nextToken());
		V = Integer.parseInt(st.nextToken());

		node = new ArrayList[N + 1];
		visitedBfs = new boolean[N + 1];
		visitedDfs = new boolean[N + 1];
		for (int i = 0; i < N + 1; i++) {
			node[i] = new ArrayList<>();
		}

		for (int o = 0; o < M; o++) {
			st = new StringTokenizer(br.readLine());
			int x = Integer.parseInt(st.nextToken());
			int y = Integer.parseInt(st.nextToken());
			node[x].add(y);
			node[y].add(x);
		}

		for (int i = 0; i < N + 1; i++) {
			node[i].sort(Integer::compareTo);
		}

		dfs(V);
		System.out.println();
		bfs();
	}

	public static void bfs() {
		Queue<Integer> q = new LinkedList<>();
		q.offer(V);
		visitedBfs[V] = true;
		System.out.print(V + " ");

		while (!q.isEmpty()) {
			int x = q.poll();
			for (int i = 0; i < node[x].size(); i++) {
				int a = node[x].get(i);
				if (!visitedBfs[a]) {
					q.offer(a);
					visitedBfs[a] = true;
					System.out.print(a + " ");
				}
			}
		}
	}

	public static void dfs(int n) {
		if (visitedDfs[n]) return;

		System.out.print(n + " ");
		visitedDfs[n] = true;
		for (int i = 0; i < node[n].size(); i++) {
			int x = node[n].get(i);
			if (!visitedDfs[x]) {
				dfs(x);
			}
		}
	}

}